src/circuit/gadget/ecc/chip/mul_fixed/full_width.rs

```diff
@@ -294,7 +295,9 @@ pub mod tests {
         {
             let scalar_fixed = pallas::Scalar::zero();
             let (result, _) = base.mul(layouter.namespace(|| "mul by zero"), Some(scalar_fixed))?;
-            assert!(result.inner().is_identity().unwrap());
+            if let Some(is_identity) = result.inner().is_identity() {
+                assert!(is_identity);
+            }
```

Comment:
We should panic on `else` here right?

Reply:
No; that is what the prior behaviour was, and it prevented the test circuit from being used in non-witness contexts like keygen or dev tooling. We can only enforce this assertion if we have a witness to enforce it on.